code/go/0chain.net/smartcontract/interestpoolsc/inter-pool.go

```diff
@@ -0,0 +1,57 @@
+package interestpoolsc
+
+import (
+	"encoding/json"
+
+	"0chain.net/chaincore/state"
+	"0chain.net/chaincore/tokenpool"
+)
+
+type interestPool struct {
+	*tokenpool.ZcnLockingPool `json:"pool"`
+	APR                       float64       `json:"apr"`
+	TokensEarned              state.Balance `json:"tokens_earned"`
+}
+
+func newInterestPool() *interestPool {
+	return &interestPool{ZcnLockingPool: &tokenpool.ZcnLockingPool{}}
+}
+
+func (ip *interestPool) encode() []byte {
+	buff, _ := json.Marshal(ip)
+	return buff
+}
+
+func (ip *interestPool) decode(input []byte) error {
+	var objMap map[string]*json.RawMessage
+	err := json.Unmarshal(input, &objMap)
+	if err != nil {
+		return err
+	}
+	ir, ok := objMap["apr"]
+	if ok {
+		var rate float64
+		err = json.Unmarshal(*ir, &rate)
+		if err != nil {
+			return err
+		}
+		ip.APR = rate
+	}
+	ie, ok := objMap["tokens_earned"]
+	if ok {
+		var earned state.Balance
```

Comment:
I know there is no generics in Golang, but can we avoid copy-pasting? In this particular place, `Balance` can be parsed as `float` and then be casted. We can create a utility module for encoding/decoding since it looks to be quite pervasive.

Reply:
Error `fatal error config file: Config File "sc" Not Found` caused because in this package there is no configuration directory.
I have already added 
`COPY ./config/sc.yaml $SRC_DIR/go/0chain.net/smartcontract/interestpoolsc/config/`
Be sure please that you can find it in your **docker.local/build.sc_unit_test/Dockerfile**

I think it makes no sense to create a separate package for encoding/decoding because there, for different structures, the encoding/decoding is done in different ways. 
And what don't you like here? )

Reply:
The error I reported above is from our CI server, not from my local machine.
If some configuration needs to be made, then it should be committed.
CI server runs unit tests with `./docker.local/bin/unit_test.sh --ci`.

About copy-paste — nevermind, I've already realized it is difficult to avoid it in Go :)

Reply:
To run the smart-contract unit test I use sc_unit_test.sh.

Reply:
We need to merge this first then:
https://github.com/0chain/0chain/pull/127